code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -569,6 +575,22 @@
 			container.lefthand_file = style["lefthand_file"]
 			container.righthand_file = style["righthand_file"]
 
+
+//Checks to see if the target reagent is being created (reacting) and if so prevents transfer
+//Only prevents products from being moved so that people can still manlipulate input reagents
+/obj/machinery/chem_master/proc/check_reactions(datum/reagent/reagent, datum/reagents/holder)
+	if(!reagent)
+		return FALSE
+	var/canMove = TRUE
+	for(var/datum/equilibrium/E in holder.reaction_list)
```

Comment:
Are there mixed types in `reaction_list` or can we assume there's only equilibrium datums here?
You can replicate this behavior with a signal if you turn it into a component.

Reply:
`reaction_list` should only ever have `/datum/equilibrium`s in it.

Reply:
```suggestion
	for(var/e in holder.reaction_list)
		var/datum/equilibrium/E = e
```
No need to type-check then.